api/v1alpha1/ocivalidator_types.go

```diff
@@ -25,47 +25,44 @@ import (
 // OciValidatorSpec defines the desired state of OciValidator
 type OciValidatorSpec struct {
 	OciRegistryRules []OciRegistryRule `json:"ociRegistryRules,omitempty" yaml:"ociRegistryRules,omitempty"`
-	EcrRegistryRules []EcrRegistryRule `json:"ecrRegistryRules,omitempty" yaml:"ecrRegistryRules,omitempty"`
 }
 
 func (s OciValidatorSpec) ResultCount() int {
-	return len(s.EcrRegistryRules) + len(s.OciRegistryRules)
+	return len(s.OciRegistryRules)
 }
 
 type OciRegistryRule struct {
-	Host            string    `json:"host" yaml:"host"`
-	RepositoryPaths []string  `json:"repositoryPaths,omitempty" yaml:"repositoryPaths,omitempty"`
-	Auth            BasicAuth `json:"auth,omitempty" yaml:"auth,omitempty"`
-	Cert            string    `json:"cert,omitempty" yaml:"cert,omitempty"`
+	// Host is a reference to the host URL of an OCI compliant registry
+	Host string `json:"host" yaml:"host"`
+
+	// Artifacts is a slice of artifacts in the host registry that should be validated.
+	// An individual artifact can take any of the following forms:
+	// <repository-path>/<artifact-name>
+	// <repository-path>/<artifact-name>:<tag>
+	// <repository-path>/<artifact-name>@<digest>
+	//
+	// When no tag or digest are specified, the default tag "latest" is used.
+	Artifacts []string `json:"artifacts" yaml:"artifacts"`
+
+	// Auth provides authentication information for the registry
+	Auth Auth `json:"auth,omitempty" yaml:"auth,omitempty"`
+
+	// CaCert is the base64 encoded CA Certificate
+	CaCert string `json:"caCert,omitempty" yaml:"caCert,omitempty"`
 }
 
 func (r OciRegistryRule) Name() string {
-	return fmt.Sprintf("%s/%s", r.Host, r.RepositoryPaths)
+	return fmt.Sprintf("%s/%s", r.Host, r.Artifacts)
 }
 
-type BasicAuth struct {
+type Auth struct {
+	// Username is the username to use for authentication.
```

Comment:
Basic creds directly in this struct is a security issue. Please replace this with `SecretName string` like we are doing for vSphere and Azure. Then the controller will have to fetch the secret and extract the creds.